barustenberg/src/ecc/curves/bn254_scalar_multiplication.rs

```diff
@@ -0,0 +1,75 @@
+use std::marker::PhantomData;
+
+use ark_bn254::{G1Affine, G1Projective};
+use ark_ec::AffineRepr;
+use ark_ff::{FftField, Field, One, Zero};
+
+pub(crate) type G1AffineGroup = <ark_ec::short_weierstrass::Affine<
+    <ark_bn254::Config as ark_ec::bn::BnConfig>::G1Config,
+> as ark_ec::AffineRepr>::Group;
+
+#[inline]
+fn cube_root_of_unity<F: ark_ff::Field>() -> F {
+    // // endomorphism i.e. lambda * [P] = (beta * x, y)
+    // if constexpr (Params::cube_root_0 != 0) {
+    //     constexpr field result{
+    //         Params::cube_root_0, Params::cube_root_1, Params::cube_root_2, Params::cube_root_3
+    //     };
+    //     return result;
+    // } else {
+    let two_inv = F::from(2_u32).inverse().unwrap();
+    let numerator = (-F::from(3_u32)).sqrt().unwrap() - F::from(1_u32);
+    two_inv * numerator
+    // constexpr field two_inv = field(2).invert();
+    // constexpr field numerator = (-field(3)).sqrt() - field(1);
+    // constexpr field result = two_inv * numerator;
+    // return result;
+    //}
+}
+pub(crate) fn is_point_at_infinity(point: &G1Projective) -> bool {
+    point.x.is_zero() && point.y.is_one() && point.z.is_zero()
```

Comment:
couldn't this just be 

`!point.x.is_zero() && !point.y.is_zero() && point.z.is.zero()`?

Reply:
it'd actually be point.x.is_zero() && !point.y.is_zero() && point.z.is.zero() but yes good catch mine was wrong

Reply:
fixed